### PR TITLE
Fix use after free error in test code

### PR DESCRIPTION
**Pull request by seelabs**

### CODE REVIEW

File: src/test/app/ValidatorSite_test.cpp [hunk 1]
```diff
@@ -97,6 +104,11 @@ class http_sync_server
     {
         error_code ec;
         acceptor_.close(ec);
```
Comment (wilsonianb) [inline src/test/app/ValidatorSite_test.cpp]: Does `close` not already protect us?

> This function is used to close the acceptor. Any asynchronous accept operations will be cancelled immediately.
> A subsequent call to open() is required before the acceptor can again be used to again perform socket accept operations. 

http://www.boost.org/doc/libs/1_64_0/doc/html/boost_asio/reference/basic_socket_acceptor/close/overload1.html

Comment (seelabs) [inline src/test/app/ValidatorSite_test.cpp]: @wilsonianb Calling `close` will cause the `on_accept` callback to be called. Before this patch, that would access the member variable `accessor_`, which may happen after the destructor has completed running. So there's definitely an issue in the existing code.

However, I just ran it and `ec` is always set in the callback when `close` is called. I should be able to get rid of the condition variable as long as I check the error code first. That will simplify this code. I'll submit a new patch shortly.

**Comment by seelabs** [comment]
Rebased onto 0.80.0-b3

File: src/test/app/ValidatorSite_test.cpp [hunk 2]
```diff
@@ -137,13 +125,10 @@ class http_sync_server
     void
     on_accept(error_code ec)
     {
+        // ec must be checked before `acceptor_` or the member variable may be
+        // accessed after the destructor has completed
         if(ec || !acceptor_.is_open())
```
Comment (miguelportilla) [inline src/test/app/ValidatorSite_test.cpp]: So previously `ec` was checked after `acceptor`, which of course is a bug.  Nice find.

**Comment by seelabs** [comment]
In 0.80.0-b4

**Pull request closed without merging**